### ISSUE TITLE
When using -pc output in the terminal, some Chinese characters cannot be displayed normally

### ISSUE THREAD

**Issue by chenqianhe** [ggerganov/whisper.cpp#399]
just like https://github.com/ggerganov/whisper.cpp/issues/25, when transcribed in zh(chinese), there are still some characters missing, and the model is from ggml-large.bin in hugging-face(https://huggingface.co/datasets/ggerganov/whisper.cpp/tree/main).

Maybe the large model and the large-v1 model still need to be repaired? 

error example:
> [00:00:13.000 --> 00:00:15.000]  各种AI的应用��出不��

ground truth:
> [00:00:13.000 --> 00:00:15.000]  各种AI的应用层出不穷

**Comment by ggerganov** [comment]
It's already converted using the solution in #25. Maybe there is some other problem with the encoding - not sure

**Comment by chenqianhe** [comment]
I'm very sorry that I may have caused some misunderstanding. Through experiments, I found that using -pc output on the terminal will lead to garbled code; And I initially determined that this was caused by the fixed length of char:
```c
const char * text = whisper_full_get_token_text(ctx, i, j);
```
Like #25, sometimes the Chinese code length should be 2. I guess this should be the problem.
And I modified the output statement and added spaces to get such results, which may prove my guess.

```c
printf("%s%s %s%s", speaker.c_str(), k_colors[3].c_str(), text, "\033[0m");
```

> [00:00:13.000 --> 00:00:15.000]   各 种 AI 的 应 用 � � 出 不 � �

Finally, I would like to apologize for the trouble that my wrong description may cause again.

**Comment by ggerganov** [comment]
No problems. Just to make sure I understand - when you remove the `-pc` flag, do you get the correct characters?

**Comment by chenqianhe** [comment]
> No problems. Just to make sure I understand - when you remove the `-pc` flag, do you get the correct characters?

Yes, it outputs normally

**Comment by chenqianhe** [comment]
> No problems. Just to make sure I understand - when you remove the `-pc` flag, do you get the correct characters?

At the same time, when I output to the file, it is normal.
Call `Whisper_full_get_segment_text(ctx, i);`  I will get the correct output

**Comment by giannhskp** [comment]
Any news on this issue? I am facing the same problem in Greek. The bug also occurs when using ```-ml``` with a small value, for example ```-ml 5```.

**Comment by chenqianhe** [comment]
> Any news on this issue? I am facing the same problem in Greek. The bug also occurs when using `-ml` with a small value, for example `-ml 5`.

I judge that this is caused by the uncertainty of the number correspondence between words and tokens. I think this will also lead to errors in obtaining the timestamp of each word. But I'm sorry, I haven't found a good way to fix this problem

**Comment by giannhskp** [comment]
I indeed intend to get word-level timestamps. This error does not occur during normal transcription (without ```-ml``` argument).

@ggerganov Any idea on what is causing this bug, or how it may be fixed?

**Comment by ggerganov** [comment]
If you add the `--split-on-word` argument does it fix the issue?

**Comment by StoneMoe** [comment]
Force windows to use UTF8 for terminal stdout encoding in Windows' language setting may fix this issue.
or whisper.cpp should detect which encoding the terminal is using.

**Comment by StoneMoe** [comment]
For me, forcing Windows to use UTF8 for terminal stdout encoding can fix this issue
or convert encoding to match current stdout encoding by a simple Python wrapper:
```Python
import subprocess
import sys
import os

codec = sys.stdout.encoding

proc = subprocess.Popen(
    f".\\whisper\\main.exe -osrt -pp -l {lang} -m model/ggml-model-whisper-large.bin -f source.wav",
    shell=True, cwd=os.getcwd(), stdout=subprocess.PIPE
)
for line in proc.stdout:
    print(line.decode(codec).rstrip())
```
For whisper.cpp, this can be fixed by detecting stdout encoding and encode output with the encoding that host is using

**Comment by bobqianic** [comment]
Fixed in #1313
